### ISSUE TITLE
Trying this tool with FNATIC Streak

### ISSUE THREAD

**Issue by hendrik-r** [XanClic/leddy#1]
I am trying to use this program on Windows 10 Home with a Fnatic Streak keyboard, but I am getting following error: 
```
leddy color=rgb:ff4000
thread 'main' panicked at 'called `Option::unwrap()` on a `None` value', /rustc/76a252ea9e7be93a61ffdf33b3533e24a9cf459d\src\libcore\macros\mod.rs:15:40
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace.
```

**Comment by XanClic** [comment]
Hi,

First of all thanks for your interest. :)

The reason for the error is that I don’t have a Streak, only a Ministreak, so leddy only looks for a Ministreak (and panics when there is none). I don’t even know whether the code actually works for a normal Streak (I suspect most of it will, but there’s going to be at least some glitches with anything that sets individual key colors, i.e. the software effects and all-keys/stdin).
I can’t find the normal Streak’s USB ID anywhere, but from a bit trial and error on a Google search, I suspect it to be 0x0101 (0x0102 is the Ministreak).

I suppose I can let leddy look for a normal Streak, too (if it is 0x0101), and then you could see for yourself what works and what doesn’t.

**Comment by XanClic** [comment]
With c98002e1d387896, leddy now looks for normal-sized Streak keyboards, too. I hope it works (I just can’t test it): https://xanclic.moe/leddy.exe

**Comment by hendrik-r** [comment]
Hey,
unfortunately I dont know how to program in rust, but if you want to add support for the Streak to your programm I would be willing to help you debugging and testing your software. 
Otherwise it looks like I have to figure it out myself.

**Comment by hendrik-r** [comment]
I tried every command except screen-capture (because i have to download ffmpeg first) and everything works without problems

**Comment by XanClic** [comment]
Great! :)

**Comment by hendrik-r** [comment]
screen-capture works almost. The screen gets mirrored for the size of the miniStreak. The numpad is one column black and 3 cols blue.

**Comment by hendrik-r** [comment]
And one quick question. How did you figure out the correct commands to send to the keyboard?

**Comment by XanClic** [comment]
> screen-capture works almost. The screen gets mirrored for the size of the miniStreak. The numpad is one column black and 3 cols blue.

OK. Interesting how some of the columns are blue. Maybe the keyboard just retains some colors for keys that are not set.
I suppose we could try to find out which key IDs they have if you’re interested in that stuff working, but that might take a bit of time.

> And one quick question. How did you figure out the correct commands to send to the keyboard?

I used Wireshark to read the USB traffic while playing around in Fnatic OP. After a while (and some trial and error), I figured out some patterns.
